app/assets/javascripts/discourse/app/helpers/share-url.js

```diff
@@ -0,0 +1,11 @@
+import { helperContext } from "discourse-common/lib/helpers";
+import User from "discourse/models/user";
+
+export function resolveShareUrl(url) {
```

Comment:
One thing that you could do is define this as `resolveShareUrl(user, url)`, and pass `this.currentUser` into the function when used in components. Then you can avoid importing `User`. 

Also I would probably scratch line 5 and just make line 7
```
const badgesEnabled = helperContext.siteSettings.enable_badges;
```

Reply:
Thanks Mark!

I wasn't able to get `this.currentUser` working but I get your concern. Since the `User` is defined in both the Topic and Post models. I've made some changes to pass the current user from there instead of importing in the helper. 

I've also made the other change your requested to remove the `siteSettings` const.